build/xocl_petalinux_compile/build.sh

```diff
@@ -0,0 +1,234 @@
+#!/bin/bash
+
+set -e
+
+error()
+{
+    echo "ERROR: $1" 1>&2
+    usage_and_exit 1
+}
+
+usage()
+{
+    echo "Usage: $PROGRAM [options] "
+    echo "  options:"
+    echo "          -help                           Print this usage"
+    echo "          -aarch                          Architecture <aarch32/aarch64/versal>"
+    echo "          -cache                          path to sstate-cache[optional]"
+    echo "          -setup                          setup file to use[optional]"
+    echo "          -clean, clean                   Remove build directories"
+    echo ""
+}
+
+usage_and_exit()
+{
+    usage
+    exit $1
+}
+
+PROGRAM=`basename $0`
+OSDIST=`grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
+BUILDDIR=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
+CMAKE=cmake
+CORE=`grep -c ^processor /proc/cpuinfo`
+jcore=$CORE
+CONFIG_FILE=""
+PETA_BSP=""
+PROJ_NAME=""
+PLATFROM=""
+clean=0
+SSTATE_CACHE=""
+SETTINGS_FILE="petalinux.build"
+
+
+if [[ $OSDIST == "centos" ]] || [[ $OSDIST == "amzn" ]] || [[ $OSDIST == "rhel" ]] || [[ $OSDIST == "fedora" ]]; then
+    CMAKE=cmake3
+    if [[ ! -x "$(command -v $CMAKE)" ]]; then
+        echo "$CMAKE is not installed, please install cmake3 by running following commands..."
+        echo "sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm"
+        echo "sudo yum install -y cmake3"
+        exit 1
+    fi
+fi
+
+release_dir="cmake_files"
+
+
+# --- Internal funtions ---
+install_recipes()
+{
+    META_USER_PATH=$1
+
+    SAVED_OPTIONS_LOCAL=$(set +o)
+    set +e
+    mkdir -p ${META_USER_PATH}/recipes-xrt/xocl
+    XOCL_BB=${META_USER_PATH}/recipes-xrt/xocl/xocl_git.bb
+    cp $BUILDDIR/xocl_git.bb $XOCL_BB
+    grep "inherit externalsrc" $XOCL_BB
+    if [ $? != 0 ]; then
+        echo "inherit externalsrc" >> $XOCL_BB
```

Comment:
Creation of a file by echo'ing and appending works fine, but looks funky when too many lines are involved. This is better IMO and easier to modify without forgetting quotes and >> etc.
```
cat << EOF > $XOCL_BB
first line
second line
...
EOF
```

Reply:
@stsoe made changes as requested, please review